collector/snapshots.go

```diff
@@ -43,6 +46,13 @@ type Snapshots struct {
 	client *http.Client
 	url    *url.URL
 
+	updateInterval time.Duration
+	lastResponse   map[string]SnapshotStatsResponse
+	lastError      error
+	lastFetchAt    time.Time
+	lock           sync.RWMutex
+	group          singleflight.Group
```

Comment:
I think it helps readability to put the lock/mutex just before the resources it is protecting and add a blank line before the lock and a blank line after the protected resources just to visually see exactly what the lock is for.